### ISSUE TITLE
[TASK] Verbose file logging

### ISSUE THREAD

**Issue by minikN** [minikN/cizn#1]
# Description
Add verbose file logging / debug flag to help users with their problems

# Description
If a user runs into a particular problem, we need a way to get insight on what the application did that caused this problem. Currently, there only is informational logging that is printed to `stdout`. This is meant for the end user to understand what the application is doing. In addition to that, we need a way to log what the application does (in as much detail as possible) to a file, so that the developers get insight on what step caused a particular problem

# Usage
This feature should only be used optionally. We could implement a global `-d, --debug` flag that enables this feature for the current execution. A log file should be created in a standard-conforming directtory (probably `/var/log/cizn/<timestamp>.log`). The debug log should not be printed to `stdout`, only to the file. In addition to that, the application should inform the user of the log file created at the end of the execution (something like `[INFO] Created detailed log at /path/to/log`)

# Implementation
We should create new adapter functions that can be fed to `tap` in order to perform this. Something like:
```js
export const initApplicationState = (app: Cizn.Application) => {
  const FS = app.Manager.FS.Api
  const logAs = logString(app)

  return asyncPipe(
    Success(app),

    // Reading and setting config
    map(getConfigPath),
    tap(logVerboseAs('info', 'Got config file: %d')) // <---
    tap(logAs('info', 'Reading config file %d...')),
    map(getConfigValues(FS)),
    tap(logVerboseAs('info', 'Config values: %d')) // <---
    // ...
  )
}
```
By default, it sohuld prints the args it's received. Additionally, we could add the current file and line to the log message to better distinguish them from one another. Logging being a side-effect, we should be explicit about it whereever we do it. By that I mean we should only do it using `tap`, as a chain link in a pipe. Other function should never do this inline.

To summarize, a verbose log message should contain these elements
- Path to file/line that called the method
- Log level
- Log message
- Timestamp
- Arguments the function was called with (e.g. the data in the pipe at that particular moment)
- Errors (including underlying js/node errors, with stack trace)

# Discussion
Maybe we should use an external logger package for this, like [winston](https://github.com/winstonjs/winston)?
